src/assets/scripts/client/aircraft/AircraftCommander.js

```diff
@@ -592,19 +575,23 @@ export default class AircraftCommander {
             taxiDestination = airport.departureRunwayModel.name;
         }
 
-        const runway = AirportController.airport_get().getRunway(taxiDestination.toUpperCase());
+        const runwayModel = AirportController.airport_get().getRunway(taxiDestination.toUpperCase());
+
+        if (!runwayModel) {
+            return [false, `unable to find Runway ${taxiDestination.toUpperCase()} on our charts`];
+        }
 
-        if (!runway) {
-            return [false, `no runway ${taxiDestination.toUpperCase()}`];
+        if (!aircraft.fms.isRunwayModelValidForSid(runwayModel)) {
+            aircraft.pilot.cancelDepartureClearance(aircraft);
         }
 
-        const readback = aircraft.pilot.taxiToRunway(runway, isDeparture, flightPhase);
+        const readback = aircraft.pilot.taxiToRunway(runwayModel, isDeparture, flightPhase);
 
         // TODO: this may need to live in a method on the aircraft somewhere
-        aircraft.fms.departureRunwayModel = runway;
+        aircraft.fms.departureRunwayModel = runwayModel;
```

Comment:
do we have a method in the `Fms` to update the runway?  I am wary about setting things like this outside the class, it makes it hard to track things down.

I understand it was already like this, buuuuuut, now's the time to tackle these.  Or maybe just update the comment, this method should live in the `Fms`.

Reply:
done